File: 746-min-cost-climbing-stairs/746-min-cost-climbing-stairs.js
/**
 * @param {number[]} cost
 * @return {number}
 */
var minCostClimbingStairs = function (cost) {
  let [a, b] = [cost[0], cost[1]];

  for (let i = 2; i < cost.length; i++) {
    const tmp = cost[i] + Math.min(a, b);
    [a, b] = [b, tmp];
  }

  return Math.min(a, b);
};

File: 746-min-cost-climbing-stairs/746-min-cost-climbing-stairs.py
class Solution:
    def minCostClimbingStairs(self, cost: List[int]) -> int:
        a, b = cost[0], cost[1]
        for i in range(2, len(cost)):
            temp = min(a, b) + cost[i]
            a, b = b, temp
        return min(a, b)